libs/features/webapp-prompt/webapp-prompt.js

```diff
@@ -43,8 +53,13 @@ class AppPrompt {
     this.getAnchorState = getAnchorState;
     this.id = this.promptPath.split('/').pop();
     this.elements = {};
-
-    this.init();
+    if (modalsActive()) {
+      window.addEventListener(
+        'milo:modal:closed',
+        () => waitForClosedModalsThen(this.init),
```

Comment:
Could we not add `waitForClosedModalsThen` as a class method and avoid passing the `init` method?

Reply:
I like to be explicit about what I'm passing to functions, and I like the CPS-ish semantics of writing "do an action and then do this - " by explicitly passing the function (so long as we aren't doing long chains of it, of course). Though if you think the `init` method shouldn't be passed I can change it (my reasoning for why it's ok is that it's not a private method; anyone can call it).